### PR TITLE
fix mass normalization

### PR DESCRIPTION
**Pull request by armingeiser**
https://www.researchgate.net/post/How_can_I_obtain_mass_normalized_mode_shapes

### CODE REVIEW

**Comment by philbucher** [comment]
Are all tests passing?
According to your source this is how it should be done right?

**Comment by armingeiser** [comment]
Yes this is how it should be done.

Tests are passing, however there is no test that tests this feature in particular. This is just scaling the eigenvectors, so it was not affecting the solution.
Is there a way to do SparseMatrix - Vector multiplication in python? that would be required to write a test...

**Comment by philbucher** [comment]
hm good question, maybe in the tests of the solvers in the core?

maybe @jcotela or @loumalouomega know sth

**Comment by armingeiser** [comment]
I found it in the tests. Thanks

**Comment by armingeiser** [comment]
ping @oberbichler

File: applications/EigenSolversApplication/tests/test_eigensystem_solver.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            "tolerance": 1e-8,
+            "normalize_eigenvectors": true,
+            "echo_level": 0
+         }''')
+
+        K = KratosMultiphysics.CompressedMatrix()
+
+        this_file_dir = os.path.dirname(os.path.realpath(__file__))
+        base_dir = os.path.dirname(os.path.dirname(os.path.dirname(this_file_dir)))
+        matrix_file_path = os.path.join(base_dir, "kratos", "tests", "A.mm")
```
Comment (oberbichler) [inline applications/EigenSolversApplication/tests/test_eigensystem_solver.py:28]: I would use the same approach as in the test template provided by Kratos:
```python
import os

def GetFilePath(fileName):
    return os.path.join(os.path.dirname(__file__), fileName)
```
and then
`GetFileName("../../../A.mm")`

File: applications/EigenSolversApplication/custom_solvers/eigensystem_solver.h [hunk 2]
```diff
@@ -275,13 +275,13 @@ class EigensystemSolver
         // eigenvector is normalized such that eigenvector^T * B * eigenvector = 1
         if(mParam["normalize_eigenvectors"].GetBool())
         {
-            if (echo_level > 0)
-                std::cout << "EigensystemSolver: Eigenvectors are normalized." << std::endl;
             for (int i = 0; i != nroot; ++i)
             {
                 const double tmp = eigvecs.row(i) * b * eigvecs.row(i).transpose();
```
Comment (oberbichler) [inline applications/EigenSolversApplication/custom_solvers/eigensystem_solver.h:280]: btw: `eigvecs.row(i).squaredNorm() * b` might be slightly faster

**Comment by oberbichler** [comment]
Sorry for the delay... looks good!

Maybe we should make a cleanup of the application after #615

**Review by oberbichler — APPROVED** [review]

**Pull request merged**